providers/ms365/resources/users.go

```diff
@@ -274,6 +293,125 @@ func (a *mqlMicrosoftUser) populateJobContactData() error {
 	return nil
 }
 
+func (a *mqlMicrosoftUser) auditlog() (*mqlMicrosoftUserAuditlog, error) {
+	res, err := CreateResource(a.MqlRuntime, "microsoft.user.auditlog", map[string]*llx.RawData{
+		"__id":   llx.StringData(a.Id.Data),
+		"userId": llx.StringData(a.Id.Data),
+	})
+	if err != nil {
+		return nil, err
+	}
+	return res.(*mqlMicrosoftUserAuditlog), nil
+}
+
+func (a *mqlMicrosoftUserAuditlog) signins() ([]interface{}, error) {
+	ctx := context.Background()
+	now := time.Now()
+	dayAgo := now.AddDate(0, 0, -1)
+	filter := fmt.Sprintf(
+		"createdDateTime ge %s and createdDateTime lt %s and (userId eq '%s' or contains(tolower(userDisplayName), '%s'))",
+		dayAgo.Format(time.RFC3339),
+		now.Format(time.RFC3339),
+		a.UserId.Data,
+		a.UserId.Data)
+	top := int32(50)
+	res := []interface{}{}
+	signIns, err := fetchUserSignins(ctx, a.MqlRuntime, filter, top)
+	if err != nil {
+		return nil, err
+	}
+	for _, s := range signIns {
+		res = append(res, s)
+	}
+	return res, nil
+}
+
+func fetchUserSignins(ctx context.Context, runtime *plugin.Runtime, filter string, top int32) ([]*mqlMicrosoftUserSignin, error) {
+	conn := runtime.Connection.(*connection.Ms365Connection)
+	betaClient, err := conn.BetaGraphClient()
+	if err != nil {
+		return nil, err
+	}
+	orderBy := "createdDateTime desc"
+	req := &auditlogs.SignInsRequestBuilderGetRequestConfiguration{
+		QueryParameters: &auditlogs.SignInsRequestBuilderGetQueryParameters{
+			Top:     &top,
+			Filter:  &filter,
+			Orderby: []string{orderBy},
+		},
+	}
+	resp, err := betaClient.AuditLogs().SignIns().Get(ctx, req)
+	if err != nil {
+		return nil, transformError(err)
+	}
+
+	res := []*mqlMicrosoftUserSignin{}
+	for _, s := range resp.GetValue() {
+		signIn, err := newMqlMicrosoftSignIn(runtime, s)
+		if err != nil {
+			return nil, err
+		}
+		res = append(res, signIn)
+	}
+	return res, nil
+}
+
+func (a *mqlMicrosoftUserAuditlog) lastInteractiveSignIn() (*mqlMicrosoftUserSignin, error) {
+	signIns := a.GetSignins()
+	if signIns.Error != nil {
+		return nil, signIns.Error
+	}
+	if len(signIns.Data) == 0 {
+		return nil, nil
+	}
+
+	latest := signIns.Data[0].(*mqlMicrosoftUserSignin)
+	return latest, nil
+}
+
+// Note: the audit log API by default excludes the non-interactive sign-ins. This is a workaround to fetch the last non-interactive sign-in.
+// We could also grab those as part of the `sign-ins` query but then the amount of data would be much larger as non-interactive logins are much more frequent.
+func (a *mqlMicrosoftUserAuditlog) lastNonInteractiveSignIn() (*mqlMicrosoftUserSignin, error) {
+	ctx := context.Background()
+	now := time.Now()
+	dayAgo := now.AddDate(0, 0, -1)
```

Comment:
Should we add this to the resource comment to highlight that this only fetches the last day?